networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriber.java

```diff
@@ -19,14 +19,18 @@
 
 import com.google.common.primitives.UnsignedLong;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Set;
+import tech.pegasys.teku.datastructures.networking.discovery.SubnetSubscription;
 import tech.pegasys.teku.networking.eth2.Eth2Network;
 import tech.pegasys.teku.util.time.channels.SlotEventsChannel;
 
 public class AttestationTopicSubscriber implements SlotEventsChannel {
-  private final Map<Integer, UnsignedLong> unsubscriptionSlotBySubnetId = new HashMap<>();
+  private final Map<Integer, UnsignedLong> allSubscriptions = new HashMap<>();
+  private final Set<Integer> persistentSubnetIdSet = new HashSet<>();
```

Comment:
Do these need to be concurrent collections?

Reply:
All of the methods of this class are synchronized, so I believe collections don't need to be concurrent.